test/apps/iodemo/io_demo.cc

```diff
@@ -1923,58 +1956,92 @@ class DemoClient : public P2pDemoCommon {
+
+                if ((delta_completed < io_op_perf_info[op].min) ||
+                    ((delta_completed == io_op_perf_info[op].min) &&
                      (server_info.retry_count >
-                      _server_info[min_index].retry_count))) {
-                    min_index = server_index;
+                              _server_info[min_index].retry_count))) {
+                    io_op_perf_info[op].min_index = server_index;
                 }
 
-                delta_min        = std::min(delta_completed, delta_min);
-                delta_max        = std::max(delta_completed, delta_max);
-                total_completed += delta_completed;
+                io_op_perf_info[op].min           =
+                        std::min(delta_completed, io_op_perf_info[op].min);
+                io_op_perf_info[op].max           =
+                        std::max(delta_completed, io_op_perf_info[op].max);
+                io_op_perf_info[op].total        += delta_completed;
+                io_op_perf_info[IO_OP_MAX].total += delta_completed;
+                server_info.prev_completed[op]    =
+                        server_info.num_completed[op];
+
+                max_min = std::max(max_min, delta_completed);
+                max     = std::max(max, delta_completed);
+            }
+
+            // Find global minimum and maximum among all operations
+            io_op_perf_info[IO_OP_MAX].min =
+                    std::min(io_op_perf_info[IO_OP_MAX].min, max_min);
+            io_op_perf_info[IO_OP_MAX].max =
+                    std::max(io_op_perf_info[IO_OP_MAX].max, max);
+        }
+
+        log << "total min:" << io_op_perf_info[IO_OP_MAX].min
+            << " max:" << io_op_perf_info[IO_OP_MAX].max
+            << " total:" << io_op_perf_info[IO_OP_MAX].total;
+
+        // Report bandwidth and min/max/total for every operation
+        for (int i = 0; i < opts().operations.size(); ++i) {
+            io_op_t op = opts().operations[i];
 
-                server_info.prev_completed[op_id] =
-                        server_info.num_completed[op_id];
+            if (total_bytes[op] == 0) {
+                continue;
             }
 
-            // Report delta of min/max/total operations for every connection
-            log << " min:" << delta_min << "(" << opts().servers[min_index]
-                << ") max:" << delta_max << " total:" << total_completed;
+            log << " | ";
+
+            double throughput_mbs = double(total_bytes[op]) /
+                                    double(UCS_MBYTE) / elapsed;
+            log << io_op_names[op] << " " << throughput_mbs << " MBs"
+                << " min:" << io_op_perf_info[op].min << "("
+                << opts().servers[io_op_perf_info[op].min_index]
+                << ") max:" << io_op_perf_info[op].max
+                << " total:" << io_op_perf_info[op].total;
+            total_bytes[op] = 0;
```

Comment:
1. can this be done with less changes?
2. can you post example of new log line?

Reply:
> I guess this is more optimal way to do it - we should go over all connections, and check operations for this connection

maybe is a little more optimal, but in this context (print once per 10 sec) simplicity is more important than performance. actually, for any non-data path code simplicity is more important than performance.

Reply:
it was necessary to inverse the order (i.e. going over all connections inside going over all connections) to find minimum of maximum number of per-connection operations done.
e.g. if we have 3 connections:
conn1: write - 5, read - 0 | min = 5, max = 5
conn2: write - 1, read - 1 | min = 1, max = 1
conn3: write - 3, read - 6 | min = 3. max = 6

total: min = 1, max - 6